python/main_loop.py

```diff
@@ -60,6 +70,10 @@ def createNewLocalPath(sailbot, maxAllowableRuntimeSeconds, desiredHeadingPublis
         rospy.logerr("Could not find a new localPath within {} seconds. Using previous localPath."
                      .format(maxAllowableRuntimeSeconds))
         return prevLocalPath, lastTimePathCreated
+
+    # Only reset createNewLocalPathAttempts if new path is successfully created
+    global createNewLocalPathAttempts
+    createNewLocalPathAttempts = 0
```

Comment:
Something to consider is to reset this every time `createNewLocalPath()` is called. Regardless if a new path is successfully created or not, the most power-intensive function `Path.createPath()` is called

Reply:
Good suggestion, I'd like to talk about this in the meeting tomorrow.

I definitely think it needs to be changed from how I have it right now, because it seems like there's a risk that `createNewLocalPathAttempts` doesn't get defined from the start if the initial local path is None.

My concern was that if we don't have a successful new path, and then 'lowWindPermitsUpdate` blocks creating another new path, could we run out of path to follow? Or maybe end up using a really bad path that puts us in danger? I thought it was safer to only reset the counter if the path we already have was "successful", ie, not None.

Reply:
We (Patrick and Jamen) discussed this on July 5th 2022, and decided to leave it as it is.

If we want to further improve power consumption, it is probably more effective and safe to increase `LOW_WIND_LOOPS_PER_PATH_UPDATE` instead of changing this. Changing this parameter makes a difference more continuously, without risking that we get stuck with a None `localPath`